src/brsTypes/components/RoSGNode.ts

```diff
@@ -96,18 +146,19 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         // method with the desired name separately? That last bit would work but it's pretty gross.
         // That'd allow roArrays to have methods with the methods not accessible via `arr["count"]`.
         // Same with RoAssociativeArrays I guess.
-        return (
-            this.elements.get(index.value.toLowerCase()) ||
-            this.getMethod(index.value) ||
-            BrsInvalid.Instance
-        );
+        let field = this.fields.get(index.value.toLowerCase());
+        let result;
+        if (field) {
+            result = field.getValue();
+        }
+        return result || this.getMethod(index.value) || BrsInvalid.Instance;
     }
 
-    set(index: BrsType, value: BrsType) {
+    set(index: BrsType, value: BrsType, alwaysNotify?: boolean) {
         if (index.kind !== ValueKind.String) {
             throw new Error("RoSGNode indexes must be strings");
         }
-        this.elements.set(index.value.toLowerCase(), value);
+        this.fields.set(index.value.toLowerCase(), new Field(value, alwaysNotify));
```

Comment:
Maybe do the null check here.

Reply:
I've changed the implementation of this, it should have the check now